services/121-service/src/payments/payments.service.ts

```diff
@@ -1222,48 +1232,40 @@ export class PaymentsService {
       );
     }
 
-    const importResults: {
-      transactions: PaTransactionResultDto[];
-      resultFeedbackPerRow: ReconciliationFeedbackDto[];
-      programFinancialServiceProviderConfigurationId: number;
-    }[] = [];
-    for await (const fspConfig of fspConfigsWithReconciliation) {
-      const importResultForFspConfig =
-        await this.importReconciliationDataPerProgramFspConfig({
-          file,
-          payment,
-          programId,
-          fspConfig,
-        });
-      const resultWithConfigId = {
-        ...importResultForFspConfig,
-        programFinancialServiceProviderConfigurationId: fspConfig.id,
-      };
-      importResults.push(resultWithConfigId);
+    const fspConfigsExcel = fspConfigsWithReconciliation.filter(
+      (c) => c.financialServiceProviderName === FinancialServiceProviders.excel,
+    );
+    let importResults;
+    if (fspConfigsExcel.length) {
+      importResults = await this.excelService.processReconciliationData({
+        file,
+        payment,
+        programId,
+        fspConfigs: fspConfigsExcel,
+      });
     }
 
-    // Store the transactions after checking each FSP config
-    for (const importResult of importResults) {
+    for (const fspConfig of fspConfigsWithReconciliation) {
       await this.transactionsService.storeAllTransactionsBulk(
```

Comment:
So we have 2 methods that are used for storing multiple transactions in the transactionService. Both are only used in the excelService since we removed vodacash. 
1. storeAllTransactionsBulk
  - Used when doing the initial send payment for excel
  - Uses objects which combine PaTransactionResultDto and TransactionRelationDetailsDto (which contains the FspConfigId) 
  - Seems more in line with current transaction-proccessor architecture
2. storeAllTransactions
  - Used when storing reconsiliation data 
  - Only works for one fspConfigId
  - Performance much better as it save in bulk

Should we remove one of these methods? And create a hybrid that accepts the parameters of storeAllTransactions but saves in bulk somehow?

Also should we test te performance of these excel imports after this refactor?